src/vs/workbench/contrib/terminal/browser/widgets/terminalQuickFixList.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IAction } from 'vs/base/common/actions';
+import { ActionList, ActionListItemKind, IListMenuItem, IRenderDelegate } from 'vs/platform/actionWidget/browser/actionWidget';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { CodeActionKind } from 'vs/editor/contrib/codeAction/common/types';
+import { Codicon } from 'vs/base/common/codicons';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+import { IActionItem } from 'vs/platform/actionWidget/common/actionWidget';
+import { localize } from 'vs/nls';
+
+export class TerminalQuickFix implements IActionItem {
+	action: IAction;
+	disabled?: boolean;
+	title?: string;
+	constructor(action: IAction, title?: string, disabled?: boolean) {
+		this.action = action;
+		this.disabled = disabled;
+		this.title = title;
+	}
+}
+
+export class QuickFixList extends ActionList<TerminalQuickFix> {
+	constructor(
+		fixes: readonly TerminalQuickFix[],
+		showHeaders: boolean,
+		delegate: IRenderDelegate<TerminalQuickFix>,
+		@IKeybindingService keybindingService: IKeybindingService,
+		@IContextViewService contextViewService: IContextViewService
+	) {
+		super('quickFixWidget', fixes, showHeaders, delegate, undefined, contextViewService, keybindingService);
+	}
+
+	public toMenuItems(inputQuickFixes: readonly TerminalQuickFix[], showHeaders: boolean): IListMenuItem<TerminalQuickFix>[] {
```

Comment:
Just building on my comment about not exposing `ActionList`, the data provider you pass the the actionWidget service can just be a class with a method that returns an `IListMenuItem<T>[]`

One other note: it would be nice if all implementers didn't have to consider `showHeaders`. Are headers specific to codeaction or does the terminal plan to use them too?